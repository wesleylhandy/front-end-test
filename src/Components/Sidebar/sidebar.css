.NotesSidebar {
	background: #a19689;
	border-right: solid 1px #636463;
	color: white;
}

.NotesSidebar__title {
	margin: 0;
	font-size: 1.25rem;
	padding: 0.2rem;
}

.NotesSidebarItem {
	padding: 0 0.3rem;
	display: flex;
	flex-direction: row;
	border-top: solid 1px #636463;
}

.NotesSidebarItem.selected {
	background: #ead592;
}

.NotesSidebarItem__title {
	margin: 0.3rem 0;
	display: inline-block;
	flex: 1;
}
