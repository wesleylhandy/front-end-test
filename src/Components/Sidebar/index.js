import React from "react";
import checkMark from "../../SVG/check-mark.svg";
import classNames from "classnames";
import PropTypes from "prop-types";
import "./sidebar.css";

const Sidebar = ({ notes = [], currentNoteIndex, selectNote }) => (
	<section className="NotesSidebar">
		<h2 className="NotesSidebar__title">Available Notes:</h2>
		<div className="NotesSidebar__list">
			{notes.length ? (
				notes.map(({ subject, read }, idx) => (
					<div
						key={`note-${idx}`}
						className={classNames("NotesSidebarItem", {
							selected: idx === currentNoteIndex,
						})}
						onClick={e => selectNote(idx)}
						id={`note-${idx}`}
						role="button"
					>
						<h4 className="NotesSidebarItem__title">{subject}</h4>
						{read && <img alt="Check Mark" src={checkMark} />}
					</div>
				))
			) : (
				<div className="NotesSidebarItem empty">--Empty--</div>
			)}
		</div>
	</section>
);

Sidebar.propTypes = {
	notes: PropTypes.arrayOf(
		PropTypes.shape({
			subject: PropTypes.string.isRequired,
			body: PropTypes.string.isRequired,
			read: PropTypes.bool.isRequired,
		})
	),
	currentNoteIndex: PropTypes.number.isRequired,
	selectNote: PropTypes.func.isRequired,
};

export default Sidebar;
