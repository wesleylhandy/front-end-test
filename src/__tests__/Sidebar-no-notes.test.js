import React from "react";
import Sidebar from "../Components/Sidebar";
import { create, act } from "react-test-renderer";

it("renders '--Empty--' when it receive 0 notes", ()=> {
    const emptyNotes = [];
	let component;
	act(() => {
		component = create(
			<Sidebar
                notes={emptyNotes}
                currentNoteIndex={-1}
                selectNote={()=>jest.fn()}
		    />
		);
	});
	component.toJSON();
	expect(component).toMatchSnapshot();
})