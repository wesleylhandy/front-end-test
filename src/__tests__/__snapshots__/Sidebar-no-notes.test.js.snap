// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders '--Empty--' when it receive 0 notes 1`] = `
<section
  className="NotesSidebar"
>
  <h2
    className="NotesSidebar__title"
  >
    Available Notes:
  </h2>
  <div
    className="NotesSidebar__list"
  >
    <div
      className="NotesSidebarItem empty"
    >
      --Empty--
    </div>
  </div>
</section>
`;
